src/event-emitter.js

```diff
@@ -28,15 +29,15 @@ define(['lodash-amd/modern/array/pull',
   EventEmitter.prototype.trigger = function (eventName, args) {
 
     //fire events like my:custom:event -> my:custom -> my
-    var events = eventName.split(':');
-    while(!!events.length){
-      var currentEvent = events.join(':');
-      var listeners = this._listeners[currentEvent] || Immutable.Set();
-      //trigger handles
-      listeners.forEach(function (listener) {
-        listener.apply(null, args);
-      });
-      events.splice((events.length - 1), 1);
+    eventName = eventName.split(':');
```

Comment:
I may go overboard sometimes when it comes to stack usage. This change will be reverted tonight. Thanks for your feedback @rrees !